workflow/controller/controller.go

```diff
@@ -311,6 +311,14 @@ func (wfc *WorkflowController) startLeading(ctx context.Context, logCtx *log.Ent
 	for i := 0; i < podWorkers; i++ {
 		go wait.Until(wfc.podWorker, time.Second, ctx.Done())
 	}
+	if v, found := os.LookupEnv("CACHE_GC_PERIOD"); found {
+		cacheGCPeriod, err := time.ParseDuration(v)
+		if err != nil {
+			log.WithField("CACHE_GC_PERIOD", v).WithError(err).Panic("failed to parse")
+		} else {
+			go wait.UntilWithContext(ctx, wfc.syncAllCacheForGC, cacheGCPeriod)
+		}
```

Comment:
maybe we could remove the else statement? 

```suggestion
		if err != nil {
			log.WithField("CACHE_GC_PERIOD", v).WithError(err).Panic("failed to parse")
			return 
		} 
		go wait.UntilWithContext(ctx, wfc.syncAllCacheForGC, cacheGCPeriod)
```